bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/watch/BidiSetBag.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.jrubyscripting.internal.watch;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+// Copy of org.openhab.core.automation.module.script.rulesupport.internal.loader.collection.BidiSetBag
```

Comment:
For my understanding, why does it need to be copied? Is it modified in any way, or just plain duplication?

Reply:
It's in an internal package and therefore not exported from the core package.

Reply:
Honestly, I'd prefer if I didn't have to copy _or_ reference it. I had to because the AbstractDependencyTracker isn't flexible enough for my needs - it assumes a single library directory that it must watch. If it were to be improved, then I could not need to reference this class at all.